logic/account.js

```diff
@@ -272,114 +270,42 @@ class Account {
 			fields = null;
 		}
 
-		const computedFieldsMap = {};
-		this.computedFields.forEach(field => {
-			computedFieldsMap[field.name] = field.dependentFields;
-		});
-
-		// If fields are not provided append computed fields
-		if (!fields) {
-			fields = this.scope.db.accounts.getDBFields();
-			fields = fields.concat(Object.keys(computedFieldsMap));
-		}
-
-		let fieldsAddedForComputation = [];
-		const performComputationFor = [];
-
-		Object.keys(computedFieldsMap).forEach(computedField => {
-			if (fields.includes(computedField)) {
-				// Add computed field to list to process later
-				performComputationFor.push(computedField);
-
-				// Remove computed field from the db fields list
-				fields.splice(fields.indexOf(computedField), 1);
-
-				// Marks fields which are explicitly added due to computation
-				fieldsAddedForComputation = fieldsAddedForComputation.concat(
-					_.difference(computedFieldsMap[computedField], fields)
-				);
-
-				// Add computation dependant fields to db fields list
-				fields = fields.concat(computedFieldsMap[computedField]);
-			}
-		});
-
-		let limit = ACTIVE_DELEGATES;
-		let offset = 0;
-		let sort = { sortField: '', sortMethod: '' };
+		const options = {
+			limit: filter.limit || ACTIVE_DELEGATES,
+			offset: filter.offset || 0,
+			sort: filter.sort || 'balance:asc',
+			extended: true,
+		};
 
-		if (filter.offset > 0) {
-			offset = filter.offset;
+		if (options.limit < 0) {
+			options.limit = ACTIVE_DELEGATES;
 		}
-		delete filter.offset;
 
-		if (filter.limit > 0) {
-			limit = filter.limit;
-		}
-		delete filter.limit;
-
-		if (filter.sort) {
-			const allowedSortFields = [
-				'username',
-				'balance',
-				'rank',
-				'missedBlocks',
-				'vote',
-				'publicKey',
-				'address',
-			];
-			sort = sortBy.sortBy(filter.sort, {
-				sortFields: allowedSortFields,
-				quoteField: false,
-			});
-		}
-		delete filter.sort;
+		const filters = _.omit(filter, ['limit', 'offset', 'sort']);
 
 		const self = this;
 
-		(tx || this.scope.db).accounts
-			.list(filter, fields, {
-				limit,
-				offset,
-				sortField: sort.sortField,
-				sortMethod: sort.sortMethod,
-			})
-			.then(rows => {
+		this.scope.storage.entities.Account.get(filters, options, tx)
+			.then(accounts => {
```

Comment:
Why not turning this into async/await?

Reply:
Just to use the same pattern already there in that function.